components/scream/src/share/tests/vertical_interp_tests.cpp

```diff
@@ -61,41 +207,41 @@ TEST_CASE("main_vertical_interpolation_test"){
 				 n_layers_src,
 				 n_layers_tgt);
 
-  Real correct_val[3][4];
-  correct_val[0][0] = 260.0;
-  correct_val[0][1] = 265.0;
-  correct_val[0][2] = 255.0;
-  correct_val[0][3] = masked_val;
-  correct_val[1][0] = masked_val;
-  correct_val[1][1] = masked_val;
-  correct_val[1][2] = 265.0;
-  correct_val[1][3] = 255.0;
-  correct_val[2][0] = 270.0;
-  correct_val[2][1] = 255.0;
-  correct_val[2][2] = masked_val;
-  correct_val[2][3] = masked_val;
-  
-  for(int col=0; col<3; col++){
-    for(int lev=0; lev<4; lev++){
+  Real correct_val[2][17];
+  for (int i=0; i<15; i++){
+    correct_val[0][i] = 205+i*5.;
+    correct_val[1][i] = 195+i*5.;
+  }
+
+  correct_val[0][14] = 255.;
+  correct_val[0][15] = 240.;
+  correct_val[0][16] = masked_val;
+  correct_val[1][0]  = masked_val;
+  correct_val[1][15] = 270.;
+  correct_val[1][16] = 255.;
+
+  for(int col=0; col<2; col++){
```

Comment:
Need to deep copy output back to host before this loop.

Reply:
Also, check mask maybe?

Reply:
Added deep copy output back to host as well as check on mask